generate-analytical-dataset.py

```diff
@@ -10,48 +10,79 @@
 from pyspark.sql.types import *
 from pyspark.sql import Row
 
+from pysecret import AWSSecret
+
+aws_profile = "analytical_dataset_generator"
+aws_region = "eu-west-2"
+aws = AWSSecret(profile_name=aws_profile, region_name=aws_region)
+
+S3_PUBLISH_BUCKET = aws.get_secret_value(
+    secret_id="ADG-Payload", key="S3_PUBLISH_BUCKET"
+)
+
+
 def main():
-	spark = SparkSession.builder.master("yarn").config("spark.sql.parquet.binaryAsString", "true").appName("spike").enableHiveSupport().getOrCreate()
-	df = spark.sql("select * from hbase_hive")
-	values = df.select("data").rdd.map(lambda x: 
-	getTuple(x.data)).map(lambda y: decrypt(y[0], y[1], y[2], y[3]))
-	print(type(values))
-	#schema = StructType(List(
-    #StructField("value", StringType(),True))
-  #)
-	#datadf = spark.createDataFrame(values,schema) 
-	row = Row("val")
-	datadf = values.map(row).toDF()
-	datadf.show()
-	datadf.write.parquet("S3_BUCKET/xxx.parquet")
-	spark.sql("""CREATE EXTERNAL TABLE hive_spark_demo1(val STRING) STORED AS PARQUET LOCATION "S3_BUCKET/source_table_parquet/xxxxx.parquet" """)
-	spark.sql("select * from hive_spark_demo1").show()
-	#getPrinted(values)
+    spark = (
+        SparkSession.builder.master("yarn")
+        .config("spark.sql.parquet.binaryAsString", "true")
+        .appName("spike")
+        .enableHiveSupport()
+        .getOrCreate()
+    )
+    df = spark.sql("select * from hbase_hive")
+    values = (
+        df.select("data")
+        .rdd.map(lambda x: getTuple(x.data))
+        .map(lambda y: decrypt(y[0], y[1], y[2], y[3]))
+    )
+    print(type(values))
+    # schema = StructType(List(
+    # StructField("value", StringType(),True))
+    # )
+    # datadf = spark.createDataFrame(values,schema)
+    row = Row("val")
+    datadf = values.map(row).toDF()
+    datadf.show()
+    datadf.write.parquet(S3_PUBLISH_BUCKET("/xxx.parquet"))
```

Comment:
I'm not sure this syntax is correct? Looking at the docs (https://pypi.org/project/pysecret/#aws-key-management-service-and-secret-manager-integration specifically), it looks like `aws.get_secret_value()` returns a string, not a function like thing. I think this might work better, or at least be a bit clearer:

`datadf.write.parquet(f"{S3_PUBLISH_BUCKET}/xxx.parquet")`